leader-election-api/src/main/java/com/palantir/leader/PingResult.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.leader;
+
+import java.util.Optional;
+
+import org.immutables.value.Value;
+
+import com.palantir.sls.versions.OrderableSlsVersion;
+
+@Value.Immutable
+public interface PingResult {
```

Comment:
Maybe premature: In practice, there are only two possible versions of this result on each TimeLock server. Thus, I'd propose interning the results to improve perf.